drivers/nrf24l01p/include/nrf24l01p_internal.h

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (C) 2019 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+/**
+ * @ingroup drivers_nrf24l01p
+ * @{
+ *
+ * @file
+ * @brief   Internal helper functions for NRF24L01P device driver
+ *
+ * @author  Hauke Petersen <hauke.petersen@fu-berlin.de>
+ * @author  Peter Kietzmann <peter.kietzmann@haw-hamburg.de>
+ * @author  Fabian Hüßler <fabian.huessler@ovgu.de>
+ */
+#ifndef NRF24L01P_INTERNAL_H
+#define NRF24L01P_INTERNAL_H
+
+#include "nrf24l01p_constants.h"
+#include "nrf24l01p.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifdef NRF24L01P_CUSTOM_HEADER
+/**
+ * @brief Indicate header byte
+ */
+#define NRF24L01P_CSTM_HDR_PREEMBLE       (0b10000000)
+/**
+ * @brief Mask of source address width
+ */
+#define NRF24L01P_CSTM_HDR_SRC_AW_MASK    (0b00000111)
+/**
+ * @brief Padding byte for ShockBurst
+ */
+#define NRF24L01P_CSTM_HDR_PADDING        (0x00)
+#endif
+/**
+ * @brief Mask of destination address width
+ */
+#define NRF24L01P_CSTM_HDR_DST_AW_MASK    (0b00111000)
+
+/**
+ * @brief   Header of a ShockBurst frame
+ */
+typedef struct {
+    /**
+     * @brief    7 6 5 4 3 2 1 0
+     *          +-+-+-+-+-+-+-+-+
+     *          |1|r|d|d|d|s|s|s|
+     *          +-+-+-+-+-+-+-+-+
+     *          r = reserved
+     *          d = destination address width
+     *          s = source address width
+     * If the protocol is ShockBurst, received frames may
+     * be padded due to static payload length. The 7th bin
+     * in the hader must be 1. Leading bytes starting with
+     * 0 are padding bytes and must be removed.
+     * Any frame having a value of ddd or sss not being in
+     * [NRF24L01P_MIN_ADDR_WIDTH; NRF24L01P_MAX_ADDR_WIDTH]
+     * shall be dropped.
+     */
+    uint8_t addr_width;
+    /**
+     * @brief   Recipient address
+     */
+    uint8_t dst_addr[NRF24L01P_MAX_ADDR_WIDTH];
+#ifdef NRF24L01P_CUSTOM_HEADER
```

Comment:
You could use `#if IS_USED(NRF24L01P_CUSTOM_HEADER)` for consistency. (Would be semantically identical to the current version.)